Cesium3DTilesSelection/src/CesiumIonTilesetLoader.cpp

```diff
@@ -492,4 +530,35 @@ CesiumIonTilesetLoader::createLoader(
             });
   }
 }
+CesiumAsync::Future<TilesetContentLoaderResult<CesiumIonTilesetLoader>>
+CesiumIonTilesetLoader::refreshTokenIfNeeded(
+    const TilesetExternals& externals,
+    const TilesetContentOptions& contentOptions,
+    int64_t ionAssetID,
+    const std::string& ionAccessToken,
+    const std::string& ionAssetEndpointUrl,
+    const AuthorizationHeaderChangeListener& headerChangeListener,
+    bool showCreditsOnScreen,
+    TilesetContentLoaderResult<CesiumIonTilesetLoader>&& result) {
+  if (result.errors.hasErrors()) {
+    std::string errorString = "Received status code 401";
+    for (std::string error : result.errors.errors) {
+      if (error.find(errorString) != std::string::npos) {
```

Comment:
Searching for some error text in the list of errors seems fragile. For example, if we change the error message slightly, it'll break silently. Can we modify `mainThreadLoadLayerJsonFromAssetEndpoint` and `mainThreadLoadTilesetJsonFromAssetEndpoint` to include the HTTP status code instead, so we can avoid this?